src/browser-frontend/views/browser/window.jsx

```diff
@@ -10,17 +10,48 @@ CONDITIONS OF ANY KIND, either express or implied. See the License for the
 specific language governing permissions and limitations under the License.
 */
 
-import React, { PureComponent } from 'react';
+import React, { PureComponent, PropTypes } from 'react';
 import CSSModules from 'react-css-modules';
+import { connect } from 'react-redux';
+import Mousetrap from 'mousetrap';
+
+import { client } from '../../global';
+import PagesModelActions from '../../actions/pages-model-actions';
+import * as SharedPropTypes from '../../model/shared-prop-types';
+import SharedActions from '../../../shared/actions/shared-actions';
+import * as UIPagesSelectors from '../../selectors/ui-pages-selectors';
+
 
 import Styles from './window.css';
 import Chrome from './chrome';
 import Content from './content';
 
+@connect(state => ({
+  client,
+  selectedPageId: UIPagesSelectors.getSelectedPageId(state),
```

Comment:
After merging this into bfg-browser (with new spinners branch) we bwinton and I were seeing all of the tabs reload on each tab change. we were wondering where it came from.